docs/Overview.md

```diff
@@ -1,96 +1,246 @@
+type Scope interface {} // ...
+type Source interface {} // ...
+type Transformation interface {} // ...
+type Value interface {} // ...
+```
 
-Below is a high level description, using the Go language, of the relations between the different components and concepts of the query engine.
+A language is used to describe a query. It consists of words whose letters are taken from an alphabet.
+Letters can be of any type and complexity, from characters, to nodes in an Abstract Syntax Tree (AST). Various languages
+can be used to represent a query.
 
 ```go
-
-type Parser interface {
-    Parse(flux string) Query
+type Language interface {
+    Name() string
 }
+```
 
-// Query defines work to be performed on time series data and a result.
-// A query is represented as a directed acyclic graph (DAG).
+A query defines a unit of work for the engine. It comprises one or more pipelines.
+It contains its representation, its results, and execution statistics.
+
+```go
 type Query interface {
-    Nodes() []QueryNode
-    Edges() []Edge
+    Results() Results
+    Stats() map[string]interface{}
+    Representation() Representation
+}
+
+type Representation interface {
+    Visitor // a representation can be visited for compilation/execution.
+    Lang() Language // a representation makes sense in a certain language.
+    String() string // a visual representation of the query.
+}
+
+// The results for a Query.
+type Results interface {
+    ForEach(func (Result))
 }
 
-// QueryNode is a single step in the DAG of a query
-type QueryNode interface {
-    ID() NodeID
-    // More details about what the node does
+type Result interface {
+    Name() string
+    // ... methods to iterate other the contents of the result
 }
+```
 
-// NodeID uniquely identifies a node.
-type NodeID
+The executor executes a query expressed in some language and fills its results, provided an execution context.
+There can be multiple executors, one for each language supported by the engine.
+The engine is responsible for providing the executor with an
+execution context that embeds everything that is necessary for the execution of the query, included the registered values,
+sources and transformations. 
 
-// Edge establishes a parent child relationship between two nodes.
-type Edge interface {
-    Parent() NodeID
-    Child() NodeID
+```go
+type Executor interface {
+    Execute(Query, ExecutionContext) error // fills query results.
+    Type() Language
 }
 
-// Planner computes a plan from a query and available storage interfaces
-type Planner interface {
-    Plan(Query, []Storage) Plan
+// The ExecutionContext contains any information necessary for properly executing a query.
+type ExecutionContext interface {
+    Bindings() Scope
+    GetSource(id string) Source
+    GetTransformation(id string) Transformation
+    
+    // ... examples of other components of the execution context.
+    Context() context.Context
+    MemoryAllocator() Allocator
+    Logger() Logger
```

Comment:
As I mentioned above, I think this interface is unnecessary. An executor should have all of the context it needs. It should not need any extra, in my opinion.